src/main/java/net/mcreator/ui/dialogs/ListSelectorDialog.java

```diff
@@ -76,8 +83,20 @@ public ListSelectorDialog(MCreator mcreator, Function<Workspace, List<T>> entryP
 		scrollPane.setBorder(BorderFactory.createEmptyBorder(0, 12, 0, 0));
 		mainComponent.add("Center", scrollPane);
 
+		if (customValue) {
+			customValueButton.addActionListener(e -> {
+				String inputValue = JOptionPane.showInputDialog(L10N.t("dialog.item_selector.enter_custom_value"));
+				if (inputValue != null && !inputValue.isEmpty()) {
+					model.addElement((T) inputValue);
+					list.setSelectedValue((T) inputValue, true);
+				}
+			});
+		}
+
 		add("Center", mainComponent);
-		add("South", PanelUtils.centerInPanel(selectButton));
+		add("South", PanelUtils.centerInPanel(customValue ?
+				PanelUtils.westAndEastElement(selectButton, customValueButton) :
+				PanelUtils.centerInPanel(selectButton)));

```

Comment:
Does `selectButton` inside another `PanelUtils#centerInpanel` change something to the UI? If it doesn't, you should simply add `selectButton` directly.

Reply:
Done!